src/app/modules/form/public-form/__tests__/public-form.controller.spec.ts

```diff
@@ -389,4 +421,741 @@ describe('public-form.controller', () => {
+      it('should return 200 with the form but without a spcpSession when the JWT token could not be found', async () => {
+        // Arrange
+        // 1. Mock the response and calls
+        const MOCK_RES = expressHandler.mockResponse()
+
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          okAsync(MOCK_SPCP_FORM),
+        )
+        MockFormService.checkFormSubmissionLimitAndDeactivateForm.mockReturnValueOnce(
+          okAsync(MOCK_SPCP_FORM),
+        )
+        MockSpcpFactory.createFormWithSpcpSession.mockReturnValueOnce(
+          errAsync(new MissingJwtError()),
+        )
+
+        // Act
+        // 2. GET the endpoint
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        // Status should be 200
+        // json object should only have form property
+        expect(MOCK_RES.json).toHaveBeenCalledWith({
+          form: MOCK_SPCP_FORM.getPublicView(),
+          isIntranetUser: false,
+          myInfoError: false,
+        })
+      })
+    })
+
+    describe('errors in form retrieval', () => {
+      it('should return 500 when a database error occurs', async () => {
+        // Arrange
+        // 1. Mock the response
+        const MOCK_RES = expressHandler.mockResponse()
+
+        // 2. Mock the call to retrieve the form
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          errAsync(new DatabaseError()),
+        )
+
+        // Act
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        // 1. Check args of mocked services
+        expect(MockAuthService.getFormIfPublic).toHaveBeenCalledWith(
+          MOCK_FORM_ID,
+        )
+        // 2. Check that error is correct
+        expect(
+          MockFormService.checkFormSubmissionLimitAndDeactivateForm,
+        ).not.toHaveBeenCalled()
+        expect(MOCK_RES.status).toHaveBeenCalledWith(500)
+      })
+
+      it('should return 404 when the form is not found', async () => {
+        // Arrange
+        // 1. Mock the response
+        const MOCK_RES = expressHandler.mockResponse()
+
+        // 2. Mock the call to retrieve the form
+        MockAuthService.getFormIfPublic.mockReturnValueOnce(
+          errAsync(new FormNotFoundError()),
+        )
+
+        // Act
+        await PublicFormController.handleGetPublicForm(
+          MOCK_REQ,
+          MOCK_RES,
+          jest.fn(),
+        )
+
+        // Assert
+        // 1. Check args of mocked services
+        expect(MockAuthService.getFormIfPublic).toHaveBeenCalledWith(
+          MOCK_FORM_ID,
+        )
+        // 2. Check that error is correct
+        expect(
+          MockFormService.checkFormSubmissionLimitAndDeactivateForm,
+        ).not.toHaveBeenCalled()
+        expect(MOCK_RES.status).toHaveBeenCalledWith(404)
```

Comment:
should we also check mockRes.body for all the errors? Just to be sure they return something in the shape we expect (`ErrorDto` shape)

Reply:
did point 1; for point 2, do you mean something like 


```ts
expect(mockResult).toBeInstanceOf(ErrorDto)
expect(mockRes.json).toBeCalledWith(mockResult)
```